Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialSystemComponent.cpp

```diff
@@ -428,6 +395,55 @@ namespace AZ
             return AzToolsFramework::AssetBrowser::SourceFileDetails();
         }
 
+        void EditorMaterialSystemComponent::OnMenuBindingHook()
+        {
+            auto menuManagerInterface = AZ::Interface<AzToolsFramework::MenuManagerInterface>::Get();
+            AZ_Assert(menuManagerInterface, "EditorMaterialSystemComponent - could not get MenuManagerInterface");
+
+            auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+            AZ_Assert(menuManagerInterface, "EditorMaterialSystemComponent - could not get ActionManagerInterface");
+
+            auto hotKeyManagerInterface = AZ::Interface<AzToolsFramework::HotKeyManagerInterface>::Get();
+            AZ_Assert(hotKeyManagerInterface, "EditorMaterialSystemComponent - could not get HotKeyManagerInterface");
+
+            constexpr AZStd::string_view ActionContext = "o3de.context.editor.mainwindow";
+            constexpr AZStd::string_view MaterialEditorAction = "o3de.tools.material_editor";
+            constexpr AZStd::string_view ToolsMenuIdentifier = "o3de.menu.editor.tools";
+
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = "Material Editor";
+            actionProperties.m_iconPath = ":/Menu/material_editor.svg";
+            auto outcome = actionManagerInterface->RegisterAction(ActionContext, MaterialEditorAction, actionProperties, [this]()
+                {
+                    OpenMaterialEditor("");
+                });
+            AZ_Assert(outcome.IsSuccess(), "Failed to RegisterAction %s", MaterialEditorAction);
+
+            outcome = menuManagerInterface->AddActionToMenu(ToolsMenuIdentifier, MaterialEditorAction, 100);
+            AZ_Assert(outcome.IsSuccess(), "Failed to AddAction %s to Menu %s", MaterialEditorAction, ToolsMenuIdentifier);
+
```

Comment:
This probably warrants a separate discussion but it seems like everything except for RegisterAction can be moved to data files or the settings registry then loaded into all of the necessary managers from a central location. I think that would shrink the code footprint and other checks wherever actions are being registered Adding action manager utility functions that wrap all of the boilerplate might also have the same impact on code footprint.

Reply:
Agreed, there is a task on the Action Manager side to centralize the global identifiers, currently they're not accessible.